### DISCUSSION TITLE
Valorizzazione dn_number

### DISCUSSION THREAD

**Discussion by dmigliacci** [fattureincloud/api#117] [Q&A] [answered]
Ciao a tutti,
sto rifattorizzando la creazione della proforma+ddt dalla v1 alla v2.
Ci sono quasi, ma ora sono bloccato in questo punto:

Nel **IssuedDocument** ho impostato il **deliveryNote** a true, ma ricevo l'errore che il numero del DDT deve essere valorizzato, esiste infatti la funzione **dnDate**.

Ma come per la proforma in cui il numero non lo gestisco io perchè è delegato alla vostra api, anche per il ddt (come nella v1) non voglio gestire la numerazione e lasciarla auto gestita dal vostro portale.

Come dovrei fare?

`{
   "error":{
      "message":"Invalid request.",
      "validation_result":{
         "dn_number":[
            "The dn number must be an integer.",
            "The dn number field is required."
         ]
      }
   }
}`

Grazie in anticipo,
Davide

**Comment by emajo** [comment]
Ciao @dmigliacci passando il `type = "delivery_note"` setta `dn_number` automaticamente, probabilmente stai ricevendo errore perché stai passando `dn_number = null` corretto?
RImuovendo il campo la numerazione verrà settata automaticamente.

Se continui a riscontrare questo problema ti consiglio di scrivere al nostro supporto clienti così possiamo andare a controllare i log delle tue chiamate per diagnosticare il problema.

Grazie,
Emanuele

**Reply by dmigliacci** [reply]
Ciao @emajo,
io non sto passando il type "**delivery_note**" bensì "**proforma**".

All'interno però è possibile (come nella v1) decidere di voler creare contestualmente anche il DDT tramite l'attributo: 

`.deliveryNote(true)`

Ma il comportamento atteso non corrisponde, quindi vorrei capire dove sto sbagliando.

Grazie,
Davide

**Reply by emajo** [reply]
Ciao @dmigliacci, ora ho capito cosa intendi ed effetivamente si verifica il comportamento da te descritto, cerchiamo di sistemare il prima possibile.

Grazie,
Emanuele

**Reply by dmigliacci** [reply]
Ok perfetto, allora attendo un vostro update al riguardo, così che posso poi andare avanti.

Grazie,
Davide

**Comment by dmigliacci** [comment]
Buongiorno,
ci sono novità riguardo questa segnalazione?
Posso fare qualcosa?

Grazie

**Comment by emajo** [comment, marked as answer]
Ciao @dmigliacci scusaci se non ti abbiamo più aggiornato sulla questione, adesso la numerazione dei ddt allegati ad altri documenti viene gestita automaticamente omettendo dn_number dalla richiesta.

Facci sapere se il tuo problema è stato risolto,
Grazie,
Emanuele

**Reply by dmigliacci** [reply]
Fantastico, a breve faccio un test e vi aggiorno.
Direi di lasciare la segnalazione aperta fino a riscontro ;-)

**Reply by dmigliacci** [reply]
it works like a charm
